plot_data/core.py

```diff
@@ -664,46 +710,39 @@ class Scatter(PlotDataObject):
+    :param elements: A list of vectors. Vectors must have the same attributes (ie the same keys)
     :param x_variable: variable that you want to display on x axis
-    :type x_variable: str
     :param y_variable: variable that you want to display on y axis
-    :type y_variable: str
-    :param tooltip: an object containing all information needed for \
-    drawing tooltips
-    :type tooltip: Tooltip
+    :param tooltip: an object containing all information needed for drawing tooltips
     :param point_style: for points' customization
-    :type point_style: PointStyle
-    :param axis: an object containing all information needed for \
-    drawing axis
-    :type axis: Axis
+    :param axis: an object containing all information needed for drawing axis
     :param log_scale_x: True or False
-    :type log_scale_x: bool
     :param log_scale_y: True or False
-    :type log_scale_y: bool
     :param heatmap: Heatmap view settings
-    :type heatmap: Heatmap
-    :param heatmap_view: Heatmap view when loading the object. If set \
-    to False, you'd still be able to enable it using the button.
-    :type heatmap_view: bool
+    :param heatmap_view: Heatmap view when loading the object.
+        If set to False, you'd still be able to enable it using the button.
     """
 
-    def __init__(self, x_variable: str, y_variable: str,
-                 tooltip: Tooltip = None,
-                 point_style: PointStyle = None,
-                 elements: List[Any] = None, axis: Axis = None,
-                 log_scale_x: bool = None, log_scale_y: bool = None,
-                 heatmap: Heatmap = None, heatmap_view: bool = None,
-                 name: str = ''):
+    def __init__(self, x_variable: str, y_variable: str, tooltip: Tooltip = None, point_style: PointStyle = None,
+                 elements: List[Sample] = None, axis: Axis = None, log_scale_x: bool = None, log_scale_y: bool = None,
+                 heatmap: Heatmap = None, heatmap_view: bool = None, name: str = ''):
         self.tooltip = tooltip
         self.attribute_names = [x_variable, y_variable]
         self.point_style = point_style
-        if not elements:
-            self.elements = []
-        else:
-            self.elements = elements
+        if elements is None:
+            elements = []
+        sampled_elements = []
+        for element in elements:
+            # RetroCompat' < 0.11.0
+            if not isinstance(element, Sample) and isinstance(element, Dict):
+                reference_path = element.pop("reference_path", "#")
+                element_name = element.pop("name", "")
+                sampled_elements.append(Sample(values=element, reference_path=reference_path, name=element_name))
+            elif isinstance(element, Sample):
+                sampled_elements.append(element)
+            else:
+                raise ValueError(f"Element of type {type(element)} cannot be used as a ScatterPlot data element.")
+        self.elements = sampled_elements
```

Comment:
Why not doing it for `plot_data.Dataset` ?

Reply:
Because I missed it ! As well as ScatterMatrix. I'll add them soon